apis/v1alpha1/gateway_types.go

```diff
@@ -500,6 +502,7 @@ type RouteNamespaces struct {
 	// * Same: Only Routes in the same namespace may be used by this Gateway.
 	//
 	// Support: Core
+	// +kubebuilder:default=Same
```

Comment:
I'm not sure how valuable a default is at this level. I can see the value of a default defined on the `RouteNamespaces` field, but if someone is going to go out of their way to define `RouteNamespaces` they'll probably want to provide a specific value here. I think the only way this default would be used is if someone specified `routeNamespaces: {}`.